### PR TITLE
Preliminary unit tests for ROS Thread Obj

### PR DESCRIPTION
**Pull request by sarimabbas**
Created some tests for ROS Thread Obj using a wrapper class containing a shared ptr. The class makes it easy to specify the number of threads to run concurrently and the thread entry functions. However, the use of pthreads means that shared ptr to void ptr inter-conversions are needed, which results in the use of raw ptrs and sacrifices the benefit (ref counts) of the smart ptr. Problem may be avoided by use of std::threads.

### CODE REVIEW

File: baxter_collaboration_lib/test/test_ros_thread_obj.test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<launch>
+    <!-- Let's load the baxter URDF from the parameter server -->
+    <include file="$(find baxter_collaboration_lib)/launch/baxter_urdf.launch" />
```
Comment (alecive) [inline baxter_collaboration_lib/test/test_ros_thread_obj.test]: Do you need the baxter URDF for this test?

File: baxter_collaboration_lib/CMakeLists.txt [hunk 2]
```diff
@@ -152,6 +152,12 @@ if(CATKIN_ENABLE_TESTING)
                                          test/test_gripper.cpp)
   target_link_libraries(test_gripper robot_interface ${catkin_LIBRARIES})
 
+  ## ROS thread obj tests
+  add_rostest_gtest(test_ros_thread_obj test/test_ros_thread_obj.test
```
Comment (alecive) [inline baxter_collaboration_lib/CMakeLists.txt]: Is this a rostest or a standard unit test in your opinion?

**Comment by alecive** [comment]
This looks good to me (see my comments though before merging). What I would do is to improve the actual test by having more behaviors with the class object you created. 

Also, I would add some more description/comments in your tests so that is easier for people to understand what is going on. After this pull request is taken care of, I would probably move the ROSThreadObj from `pthread` to `std::thread`.

Also, please review the codacy issue that says that the quality of your code can be improved (it's a very minor thing, but you can add a setting in sublime to automatically place a newline at the end of each file. It's good for the compiler, but don't ask me why :sunglasses: )

**Comment by sarimabbas** [comment]
Ok, I tried to make some of the changes you mentioned. I changed to a gtest in CMakeLists, and then added some more tests and commented the code. With the concurrent behavior, its a bit tricky to test for the end value of the variable (sometimes it becomes something random), but I think the tests inside are repeatable. Hopefully when making the switch to std::threads and the shared ownership advantage of the shared ptr, the behavior may be more consistent.

**Comment by sarimabbas** [comment]
Travis seems to have gotten stuck , i am unable to get the build to start despite making some trivial commit, and closing/reopening the pull request :(

**Comment by alecive** [comment]
Don't try to do strange things to make the build successful.. Read the details and understand why it fails!!!

**Comment by alecive** [comment]
> With the concurrent behavior, its a bit tricky to test for the end value of the variable (sometimes it becomes something random)

If it fails because the behavior is non deterministic, you should fix it, not try it again until it (randomly) succeeds :)

**Comment by sarimabbas** [comment]
Sorry!! I got confused, but now i'm following a lead on how to fix it

**Comment by sarimabbas** [comment]
I messed around with the branch a bit too much, so I'll just create a new one with the needed changes if thats ok.

**Comment by alecive** [comment]
> I messed around with the branch a bit too much, so I'll just create a new one with the needed changes if thats ok.


Okay, but it was okay also to stick with this one. Anything can be committed/reviewed :)

**Pull request closed without merging**